### ISSUE TITLE
[vo/gpu-next/vulkan/libplacebo] Imported object offset 128 conflicts with alignment 256!

### ISSUE THREAD

**Issue by Obegg** [mpv-player/mpv#9501]
### Important Information

Provide following Information:
- mpv version [0.34.0-31-gc478a1efba](https://sourceforge.net/projects/mpv-player-windows/files/64bit/mpv-x86_64-20211114-git-c478a1e.7z/download)
- Windows Version 10 Pro 21H1 19043.1348
- Source of the mpv binary [shinchiro sourceforge](https://sourceforge.net/projects/mpv-player-windows/files/64bit/)
- If known which version of mpv introduced the problem: No
- Possible screenshot or video of visual glitches: No

### Reproduction steps
Mainly because of `vo=gpu-next`

But here's the config:
```profile=gpu-hq
hr-seek-framedrop=no
no-input-default-bindings
keep-open=yes
fs
osc=yes
osd-bar=yes
osd-on-seek=msg-bar
osd-font-size=22
osd-bar-align-y=-1
osd-border-size=1
osd-bar-h=2
osd-bar-w=60
cache=yes
demuxer-lavf-probe-info=yes
demuxer-lavf-probescore=100
demuxer-lavf-probesize=999999999
demuxer-lavf-analyzeduration=3000
demuxer-lavf-linearize-timestamps=no
demuxer-mkv-subtitle-preroll=yes
demuxer-mkv-probe-video-duration=yes
deinterlace=no
hwdec=auto
vo=gpu-next
framedrop=vo
gpu-api=auto
d3d11-exclusive-fs=yes
video-sync=display-resample
blend-subtitles=no
scaler-resizes-only
scale=ewa_lanczos
cscale=ewa_lanczos
dscale=ewa_lanczos
tscale=lanczos
scale-antiring=0.0
cscale-antiring=0.0
dscale-antiring=0.0
tscale-antiring=0.0
correct-downscaling
linear-downscaling=no
deband=no
hdr-compute-peak=no
gamut-clipping=no
opengl-pbo
alang=jpn
volume-max=100
audio-exclusive=yes
audio-file-auto=all
audio-pitch-correction=yes
audio-spdif=ac3,dts,dts-hd,eac3,truehd
audio-channels=7.1,5.1,stereo
audio-stream-silence=no
slang=eng,en,en-US
embeddedfonts=yes
sub-auto=all
sub-fix-timing=no
sub-ass-scale-with-window=no
sub-ass-vsfilter-blur-compat=no
sub-ass-vsfilter-aspect-compat=no
screenshot-format=png
screenshot-high-bit-depth=yes
screenshot-png-compression=0
screenshot-jpeg-quality=100
screenshot-directory="~~desktop/"
screenshot-template="%f-%wH.%wM.%wS.%wT-#%#00n"
hls-bitrate=max
```

### Expected behavior
No errors or spam on log file.

### Actual behavior
Non ending errors on log file:
`[d][vo/gpu-next/vulkan/libplacebo] Imported object offset 128 conflicts with alignment 256!`

### Sample files
Any local file.

**Comment by haasn** [comment]
Hi, this should already be fixed in haasn/libplacebo@8cc4da8f. Please upgrade to that version, and re-open if the issue persists.

**Comment by ObsessedHacker** [comment]
> Hi, this should already be fixed in [haasn/libplacebo@8cc4da8](https://github.com/haasn/libplacebo/commit/8cc4da8f). Please upgrade to that version, and re-open if the issue persists.

upgraded to latest version and issue persists, well, kind of.
using `mpv --no-config --vo=gpu-next --gpu-context=winvk` still shows `[v][vo/gpu-next/vulkan/libplacebo] (Re)creating 960x540x0 texture with format r8` many times.
[(Re)creating 960x540x0 texture with format r8.txt](https://github.com/mpv-player/mpv/files/7613417/Re.creating.960x540x0.texture.with.format.r8.txt)

**Comment by ObsessedHacker** [comment]
Here's the log without `--no-config`, still same issue `[d][vo/gpu-next/vulkan/libplacebo] Imported object offset 128 conflicts with alignment 256!`
[Latest with same errors.txt](https://github.com/mpv-player/mpv/files/7613437/Latest.with.same.errors.txt)

**Comment by haasn** [comment]
Can you try with `--msg-level=vo/gpu-next/vulkan=trace`?

**Comment by ObsessedHacker** [comment]
> Can you try with `--msg-level=vo/gpu-next/vulkan=trace`?

Yes.
`mpv --no-config --vo=gpu-next --gpu-context=winvk --msg-level=vo/gpu-next/vulkan=trace`
[msg-level=vogpu-nextvulkan=trace.txt](https://github.com/mpv-player/mpv/files/7613598/msg-level.vogpu-nextvulkan.trace.txt)

**Comment by haasn** [comment]
That log doesn't have the error. (I assume it's caused by `--hwdec=auto`?)

**Comment by ObsessedHacker** [comment]
> That log doesn't have the error. (I assume it's caused by `--hwdec=auto`?)

I guess you are right, I did not see the error either, I guess it's because of `--hwdec=auto`, yes, now I see it.
`mpv --no-config --vo=gpu-next --gpu-context=winvk --msg-level=vo/gpu-next/vulkan=trace --hwdec=auto`
[log.txt](https://github.com/mpv-player/mpv/files/7613606/log.txt)

**Comment by haasn** [comment]
Hmm. I'm confused about where this alignment requirement is coming from. In the code, the only paths that adjust the alignment in any way are when using it as a uniform buffer, storage buffer, or texel buffer. But the usage path doesn't have any of these flags set??

This makes no sense. Can you try `(gdb) break vk_buf_create if params->import_handle`, `watch mparams.reqs.alignment`, and then stepping through the code to see where and why it changes from 1 to 256?

**Comment by ObsessedHacker** [comment]
> Hmm. I'm confused about where this alignment requirement is coming from. In the code, the only paths that adjust the alignment in any way are when using it as a uniform buffer, storage buffer, or texel buffer. But the usage path doesn't have any of these flags set??
> 
> This makes no sense. Can you try `(gdb) break vk_buf_create if params->import_handle`, `watch mparams.reqs.alignment`, and then stepping through the code to see where and why it changes from 1 to 256?

I'm sorry I didn't quite get it, can you dumb it down for me? What do you want me to do and how do I do it?
If I had to guess the issue is probably the RAM I have, it's DDR3 8GB (1 stick) 1333Mhz, and MPV `hwdec=auto` is `copy-back` (don't know why, but it does use copy-back as clearly shown in the logs), 
I did not have this issue with `hwdec=d3d11va`, but that's not supported so I can't use it.

**Comment by ObsessedHacker** [comment]
Also - I don't think I can re-open the issue, I don't have this button, can you re-open it?

**Comment by haasn** [comment]
> I'm sorry I didn't quite get it, can you dumb it down for me? What do you want me to do and how do I do it?

Not sure how to simplify it further - I'm guessing you're not familiar with C development, and conversely I'm not familiar with windows debugging. Can you maybe instead apply custom patches to libplacebo? (That might be easier)

In either case, it should be noted that, while I would still like to understand what exactly is going on here, this error is mostly cosmetic. If you just ignore the message (which only happens at `-v -v ` verbosity) everything should still be working fine, albeit with reduced performance.

Assuming the code is doing the right thing (and I just don't understand how it comes to that conclusion), then there would be no way around the performance hit anyways.

**Comment by Obegg** [comment]
>I'm guessing you're not familiar with C development
>
You are correct.

I leave the issue opened since it does seem to be an issue with ```vo=gpu-next``` + ```hwdec=dxva2-copy```.
But I no longer use ```vo=gpu-next``` with ```hwdec=auto``` (```dxva2-copy```).
Currently, I use it with ```hwdec=no```.
Waiting for ```hwdec=d3d11va``` will be implemented.

**Comment by ValZapod** [comment]
Can you restest? I think I also triggered it once or twice, but it is gone now?

**Comment by haasn** [comment]
Is this still an issue?

**Comment by Obegg** [comment]
I don't know, I'm not using 'dxva2-copy', so I will close it for now.
